src/image_gen/imagegen_init.cpp

```diff
@@ -87,7 +240,17 @@ std::variant<Status, ImageGenPipelineArgs> prepareImageGenPipelineArgs(const goo
             SPDLOG_LOGGER_ERROR(modelmanager_logger, "Default resolution exceeds maximum allowed resolution: {} > {}", args.defaultResolution.value(), args.maxResolution);
             return StatusCode::DEFAULT_EXCEEDS_MAXIMUM_ALLOWED_RESOLUTION;
         }
+        // default resolution is not among the ones allowed
+        if (args.staticReshapeSettings.has_value()) {
+            auto& resolutions = args.staticReshapeSettings.value().resolution;
+            auto it = std::find(resolutions.begin(), resolutions.end(), args.defaultResolution.value());
+            if (it == resolutions.end()) {
+                SPDLOG_LOGGER_ERROR(modelmanager_logger, "Default resolution {} is not among the static resolutions: {}", args.defaultResolution.value(), resolutions);
+                return StatusCode::SHAPE_WRONG_FORMAT;
+            }
+        }
```

Comment:
Could you explain this part? Does user specify default or has to provide resolutions with default among them?

Reply:
There is:
- resolution - where user can specify single or multiple resolutions (for single, it will enforce staticness of entire pipeline)
- max_resolution - that limits which requests can be accepted; this is useless, when `resolution` is specified and administrator gets error immediately
- default_resolution - in case there are multiple defined, or none (any resolution), this can override internal GenAI behavior of selecting default resolution. In short, user doesnt request specific size, but we can decide which one to pick